app/integration_action.go

```diff
@@ -31,90 +31,138 @@ import (
+		// Get action metadata from the database
+		post := result.Data.(*model.Post)
 
-	clientTriggerId, _, err := request.GenerateTriggerId(a.AsymmetricSigningKey())
-	if err != nil {
-		return "", err
+		result = <-cchan
+		if result.Err != nil {
+			return "", result.Err
+		}
+		channel := result.Data.(*model.Channel)
+
+		action := post.GetAction(actionId)
+		if action == nil || action.Integration == nil {
+			return "", model.NewAppError("DoPostAction", "api.post.do_action.action_id.app_error", nil, fmt.Sprintf("action=%v", action), http.StatusNotFound)
+		}
+
+		upstreamRequest.ChannelId = post.ChannelId
+		upstreamRequest.TeamId = channel.TeamId
+		upstreamRequest.Type = action.Type
+		upstreamRequest.Context = action.Integration.Context
+		datasource = action.DataSource
+
+		retainPropKeys := []string{"override_username", "override_icon_url"}
+		for _, key := range retainPropKeys {
+			value, ok := post.Props[key]
+			if ok {
+				retain[key] = value
+			} else {
+				remove = append(remove, key)
+			}
+		}
+
+		if post.RootId == "" {
+			rootPostId = post.Id
+		} else {
+			rootPostId = post.RootId
+		}
+
+		upstreamURL = action.Integration.URL
 	}
 
-	if action.Type == model.POST_ACTION_TYPE_SELECT {
-		request.DataSource = action.DataSource
-		request.Context["selected_option"] = selectedOption
+	if upstreamRequest.Type == model.POST_ACTION_TYPE_SELECT {
+		if selectedOption != "" {
+			if upstreamRequest.Context == nil {
+				upstreamRequest.Context = map[string]interface{}{}
+			}
+			upstreamRequest.DataSource = datasource
+			upstreamRequest.Context["selected_option"] = selectedOption
+		}
 	}
 
-	resp, err := a.DoActionRequest(action.Integration.URL, request.ToJson())
-	if err != nil {
-		return "", err
+	clientTriggerId, _, appErr := upstreamRequest.GenerateTriggerId(a.AsymmetricSigningKey())
+	if appErr != nil {
+		return "", appErr
 	}
 
+	resp, appErr := a.DoActionRequest(upstreamURL, upstreamRequest.ToJson())
+	if appErr != nil {
+		return "", appErr
+	}
 	defer resp.Body.Close()
 
 	var response model.PostActionIntegrationResponse
 	if err := json.NewDecoder(resp.Body).Decode(&response); err != nil {
 		return "", model.NewAppError("DoPostAction", "api.post.do_action.action_integration.app_error", nil, "err="+err.Error(), http.StatusBadRequest)
 	}
 
-	retainedProps := []string{"override_username", "override_icon_url"}
-
 	if response.Update != nil {
 		response.Update.Id = postId
 		response.Update.AddProp("from_webhook", "true")
-		for _, prop := range retainedProps {
-			if value, ok := post.Props[prop]; ok {
-				response.Update.Props[prop] = value
-			} else {
-				delete(response.Update.Props, prop)
-			}
+		for key, value := range retain {
+			response.Update.AddProp(key, value)
+		}
+		for _, key := range remove {
+			delete(response.Update.Props, key)
 		}
-		if _, err := a.UpdatePost(response.Update, false); err != nil {
-			return "", err
+		if _, appErr = a.UpdatePost(response.Update, false); appErr != nil {
```

Comment:
When the replied post is a ephemeral post, this call fails. Hence integrations are not able to update there posts via `PostActionIntegrationResponse.Update`.

Is this something we just don't want to support? Or not jet?